Sources/Purchasing/OfferingsManager.swift

```diff
@@ -80,9 +81,24 @@ class OfferingsManager {
             switch result {
             case let .success(response):
                 self.handleOfferingsBackendResult(with: response,
+                                                  appUserID: appUserID,
                                                   fetchPolicy: fetchPolicy,
                                                   completion: completion)
 
+            case let .failure(.networkError(networkError)) where networkError.isServerDown:
```

Comment:
ℹ️  This is the meat of this PR: special case for these errors: instead of forwarding the error, we try to load from disk cache. If that still fails, we return the error anyway.

Reply:
Mentioned in Linear but I wonder if we should just use the cache (if available) in all errors... While I agree that users won't be able to purchase anyway if there is no connection, I think we can still help showing the paywall with the cached data so the issue is delayed... I think this alternative is also acceptable though.

Reply:
Bump to this comment.

Reply:
Oh sorry I missed it.

I'm not sure about _all_ errors though. What about "offerings are empty"?
If you're working on your app, and break offerings, it might be better to show "offerings are empty" rather than showing the previously cached ones, so you know a fresh install won't show them.
Maybe we can do it for all network errors only? /cc @aboedo

Reply:
Merging this for now, since it solves the issue of being able to load it when the server is down.
Happy to extend to other errors in a separate PR.

Reply:
yeah, Offerings is conceptually different than CustomerInfo for this. I think only serving from cache on 5xx is the safest approach in that it's also the closest to current behavior